### ISSUE TITLE
Invalid debuginfo scope for enum of functions

### ISSUE THREAD

**Issue by rikkimax** [ldc-developers/ldc#3272]
```
invalid scope
!122 = distinct !DISubprogram(name: "__lambda2!(SomethingA, SomethingB, SomethingC)", linkageName: "\01_D6test1722Execute_Functions_Enum__T9__lambda2TCQBu10SomethingATSQCl10SomethingBTSQDc10SomethingCZQCkFQCdKQBqKQBdZv", scope: <vscale x 140696332874672 x 0x00007FF66932F3C2 (0x0000000000000001 0x00007FF6693185E7 0x00004E223B3A9773 0x0000000000000000)
```

Is proceeded by a stack trace (omitted for obvious reasons, quite long hex addresses).

This can be reproduced via (``ldc2 -g -run test.d``):

```d
interface SomethingA {}
struct SomethingB {}
struct SomethingC {}

void main()
{
    SomethingB b;
    SomethingC c;
    Execute_Functions_Enum.First(null, b, c);
}

enum Execute_Functions_Enum : void function(SomethingA, ref SomethingB, ref SomethingC) {
    First = (SomethingA, ref SomethingB, ref SomethingC) {
    }
}
```

dmd is unaffected, ldc is affected as of 1.18.0 and 1.19.0.
Platform non-specific. Tested on Windows and on run.dlang.io which is linux.

**Comment by kinke** [comment]
Thx for the interesting test case. While this uncovers LDC-specific bugs, this reduced test case segfaults with DMD too:
```D
enum E : void function()
{
    First = () {} // `= function() {}` works
}

void main()
{
    E.First();
}
```

**Comment by kinke** [comment]
Related LDC-specific bugs should be fixed by #3274; I've filed a DMD issue for the remaining frontend bug: https://issues.dlang.org/show_bug.cgi?id=20501

**Comment by rikkimax** [comment]
Thanks!
